diesel/src/macros/mod.rs

```diff
@@ -69,7 +69,13 @@ macro_rules! __diesel_column {
         {
         }
 
-        impl $crate::expression::NonAggregate for $column_name {}
+        impl $crate::expression::ValidGrouping<()> for $column_name {
+            type IsAggregate = $crate::expression::is_aggregate::No;
+        }
+
+        impl $crate::expression::ValidGrouping<$column_name> for $column_name {
```

Comment:
So if we would like to provide support for more complex `GROUP BY` clauses like `GROUP BY column_a, column_by` we need to provide an additional impl here that basically says: `impl<G> ValidGrouping<G> for $column_name where G: Contains<$column_name>` right? My question now is how would we write the required impls of `Contains<$column_name>` without some kind of specialization? As far as I can see they will conflict with each other.

(That's not an issue with this PR, I just want to understand the coarse plan, how the final feature looks like)

Reply:
From the corresponding discussion on gitter:

> An idea would be to have some proc macro attribute that applies globally to the schema.rs file and that just collects all columns and tables in this file. Then it can generate all required brute force impls. (This would also remove the need for allow_tables_to_appear_in_same_query!)